src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -42,8 +42,9 @@ class AccessDecisionManager implements AccessDecisionManagerInterface
      */
     public function __construct($voters = [], $strategy = self::STRATEGY_AFFIRMATIVE, $allowIfAllAbstainDecisions = false, $allowIfEqualGrantedDeniedDecisions = true)
     {
+        $strategy = (string) $strategy;
```

Comment:
This is not needed. The `$strategy` argument is a string.

Reply:
The `bool` arguments are also converted to `bool`. Until the code base is migrated to scalar type hints (or composition), I’d keep the cast.

Reply:
We usually do not do that anywhere else (the `bool` type cast is probably a relict from the past). The `@param` annotation already documents the expected type and we expect that this is respected by the caller.

Reply:
indeed, type cast should be avoided.